Второй этап домашней работы

SleeplessShu opened the pull request:
- Окно поиска
- Сохранение текста при повороте
- Доделанная страница настроек
- Строки в ресурсах
- Темы и стили

SleeplessShu committed:
Добавлены
 - шрифты
 - ночная тема

 проблемы:
 - изменения цвета стрелки "назад"
 - не получилась реализация switch через appcompat
 - выровнять свитч, как на рефе по линии с иконками под ним

SleeplessShu committed:
Исправлено:
- смена цвета кнопки "НАЗАД" при влкючении ночной темы работает корректно белый -> чёрный

SleeplessShu committed:
Исправлено:
- кнопка switch switch->appcompat, ночная и дневная темы работают корректно.

SleeplessShu committed:
Добавлено в SettingsActivity:
- Реализовать кнопку «Поделиться приложением»
- Реализовать кнопку «Написать в поддержку»
- Реализовать кнопку «Пользовательское соглашение»

SleeplessShu committed:
Добавлено в SettingsActivity, MainActivity, SearchActivity, MediatekaActivity:
- onResume()
- onPause()
- onStop()
- onDestroy()

SleeplessShu committed:
Добавлено:
- Окно поискового запроса
- TextWatcher
- Дополнительные цвета
Исправлено:
- Поведение переключателя темы при возврате в настройки (не запоминал положение)

SleeplessShu committed:
Добавлено:
- Функция сохранения текста в окне поиска при повороте экрана

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/MainActivity.kt:
```diff
@@ -26,11 +27,30 @@ class MainActivity : AppCompatActivity() {
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
```
Для чего это всё? Надо бы удалить. Далее аналогично

SleeplessShu replied:
Огромное спасибо за крутые комментарии.

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/MainActivity.kt:
```diff
@@ -26,11 +27,30 @@ class MainActivity : AppCompatActivity() {
+            startActivity(intentSearch)
+        }
+        val prefs = getSharedPreferences("Settings", MODE_PRIVATE)
+        val isNightModeOn = prefs.getBoolean("NightMode", false)
```
"NightMode" и вообще строки-ключи стоит выносить в отдельную константу (в данном случае в companion object). Это позволит не бояться забыть поменять в одном месте и забыть в другом. При необходимости просто меняешь в самой константе и всё так же работает. Плюс сразу понятно, где мы передаем и получаем эти данные.

SleeplessShu replied:
сделал

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,61 @@
+    <style name="ButtonsStyle">
+        <item name="android:layout_width">match_parent</item>
+        <item name="android:layout_height">match_parent</item>
+        <item name="android:layout_weight">1</item>
```
Лучше внешние отступы, высоту и ширину, веса, гравити не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи и вообще только за то, что находится внутри нее - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

SleeplessShu replied:
сделал

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/SearchActivity.kt:
```diff
@@ -14,5 +19,41 @@ class SearchActivity : AppCompatActivity() {
+            override fun afterTextChanged(s: Editable?) {
+            }
+        })
+    }
```
А это зачем, если не используется? Эта штука нужна, чтобы в onTextChanged сохранять введенный текст в переменную

SleeplessShu replied:
переделал

SleeplessShu committed:
Исправлено:
- кнопки mainActivity наследуются от Widget.MaterialComponents.Button.UnelevatedButton
- внешние отступы, высоту и ширину, веса, гравити вынес из стилей (тут у меня есть вопрос про однотипность: в мэйне, в сеттингах, получается большое количество повторяемого кода для каждой кнопки. Неужели это лучше перемещения в стили?)
- убраны оверрайд функции обращающиеся к супер классу
- переработан стиль и реализация верстки активити "Поиск"
- при нажатии на кнопку “очистить”, клавиатура скрывается
- keys вынесены в companion object
